nixos/modules/services/networking/fedimintd.nix

```diff
@@ -0,0 +1,357 @@
+            if cfg.p2p.openFirewall then [
+              cfg.p2p.port
+            ] else [ ]
+          )
+        )
+        eachFedimintd);
+
+
+    systemd.services =
+      mapAttrs'
+        (fedimintdName: cfg: (
+          nameValuePair "fedimintd-${fedimintdName}" (
+            let
+              startScript = pkgs.writeShellScript "fedimintd-start" (
+                (if cfg.bitcoin.rpc.secretFile != null then
+                  ''
+                    secret=$(${pkgs.coreutils}/bin/head -n 1 "${cfg.bitcoin.rpc.secretFile}")
+                    prefix="''${FM_BITCOIN_RPC_URL%*@*}"  # Everything before the last '@'
+                    suffix="''${FM_BITCOIN_RPC_URL##*@}"  # Everything after the last '@'
+                    FM_BITCOIN_RPC_URL="''${prefix}:''${secret}@''${suffix}"
+                  ''
+                else
+                  "") +
+                ''
+                  exec ${cfg.package}/bin/fedimintd
+                ''
+              );
+            in
+            {
+              description = "Fedimint Server";
+              documentation = [ "https://github.com/fedimint/fedimint/" ];
+              wantedBy = [ "multi-user.target" ];
+              environment = lib.mkMerge ([
+                {
+                  FM_BIND_P2P = "${cfg.p2p.bind}:${builtins.toString cfg.p2p.port}";
+                  FM_BIND_API = "${cfg.api.bind}:${builtins.toString cfg.api.port}";
+                  FM_P2P_URL = cfg.p2p.address;
+                  FM_API_URL = cfg.api.address;
+                  FM_DATA_DIR = cfg.dataDir;
+                  FM_BITCOIN_NETWORK = cfg.bitcoin.network;
+                  FM_BITCOIN_RPC_URL = cfg.bitcoin.rpc.address;
+                  FM_BITCOIN_RPC_KIND = cfg.bitcoin.rpc.kind;
+                }
+                cfg.extraEnvironment
+              ]);
+              serviceConfig = {
+                User = cfg.user;
+                Group = cfg.group;
+
+                Restart = "always";
+                RestartSec = 10;
+                StartLimitBurst = 5;
+                UMask = "077";
+                LimitNOFILE = "100000";
+
+                LockPersonality = true;
+                ProtectClock = true;
+                ProtectControlGroups = true;
+                ProtectHostname = true;
+                ProtectKernelLogs = true;
+                ProtectKernelModules = true;
+                ProtectKernelTunables = true;
+                PrivateMounts = true;
+                RestrictAddressFamilies = [ "AF_INET" "AF_INET6" ];
+                RestrictNamespaces = true;
+                RestrictRealtime = true;
+                SystemCallArchitectures = "native";
+                SystemCallFilter = [
+                  "@system-service"
+                  "~@privileged"
+                ];
+                StateDirectory = "fedimintd-${fedimintdName}";
+                StateDirectoryMode = "0700";
+                ExecStart = startScript;
+
+                # Hardening measures
```

Comment:
```suggestion
```
there are also hardening options above this comment

Reply:
Just removed and sorted it all.